src/linear_algebra/basic.lean

```diff
@@ -1977,6 +1977,30 @@ section add_comm_monoid
 variables [semiring R] [add_comm_monoid M] [add_comm_monoid M₂]
 [add_comm_monoid M₃] [add_comm_monoid M₄]
 
+section subsingleton
+variables [module R M] [module R M₂] [subsingleton M] [subsingleton M₂]
+
+/-- Between two zero modules, the zero map is an equivalence. -/
+instance : has_zero (M ≃ₗ[R] M₂) :=
+⟨{ to_fun := 0,
+   inv_fun := 0,
+   right_inv := λ x, subsingleton.elim _ _,
+   left_inv := λ x, subsingleton.elim _ _,
+   ..(0 : M →ₗ[R] M₂)}⟩
+
+-- Even though these are implied by `subsingleton.elim` via the `unique` instance below, they're
+-- nice to have as `rfl`-lemmas for `dsimp`.
+@[simp] lemma zero_symm : (0 : M ≃ₗ[R] M₂).symm = 0 := rfl
+@[simp] lemma coe_zero : ⇑(0 : M ≃ₗ[R] M₂) = 0 := rfl
```

Comment:
I'm assuming you can't do `@[simps]` to autogenerate `zero_symm` and/or `coe_zero`?

Reply:
`simps` would generate `⇑(0 : M ≃ₗ[R] M₂).symm = 0` not `(0 : M ≃ₗ[R] M₂).symm = 0`. I could use it to generate `coe_zero`, but I'm not sure it's really worth either the extra build cycle, or the inconsistency with the other two lemmas.